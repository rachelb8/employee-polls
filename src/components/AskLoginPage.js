import { useNavigate } from "react-router-dom";
import { MDBBtn } from "mdb-react-ui-kit";
import loginImage from "../icons/loginImage.png";

const AskLoginPage = () => {
  const navigate = useNavigate();

  const handleClick = (e) => {
    e.preventDefault();

    navigate("/login");
  };

  return (
    <div className="text-center m-4 d-flex align-items-center justify-content-center">
      <div style={{ width: "48rem" }}>
        <h1>Employee Polls</h1>
        <img src={loginImage} className="image" alt="login" />
        <h2>Please login to view your polls.</h2>
        <MDBBtn onClick={handleClick}>Let's login!</MDBBtn>
      </div>
    </div>
  );
};

export default AskLoginPage;
