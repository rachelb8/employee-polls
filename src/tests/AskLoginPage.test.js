import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { BrowserRouter as Router } from "react-router-dom";
import middleware from "../middleware";
import reducer from "../reducers";
import AskLoginPage from "../components/AskLoginPage";

const store = createStore(reducer, middleware);

describe("AskLoginPage", () => {
  it("matches the snapshot", () => {
    var component = render(
      <Provider store={store}>
        <Router>
          <AskLoginPage />
        </Router>
      </Provider>
    );

    var loginButton = component.getByTestId("login-button");
    fireEvent.click(loginButton);
    expect(component).toMatchSnapshot();
  });

  it("ensures login button is displayed and enabled", () => {
    var component = render(
      <Provider store={store}>
        <Router>
          <AskLoginPage />
        </Router>
      </Provider>
    );

    var loginButton = component.getByTestId("login-button");
    fireEvent.click(loginButton);
    expect(loginButton).toBeInTheDocument;
    expect(loginButton).toBeEnabled;
  });
});
