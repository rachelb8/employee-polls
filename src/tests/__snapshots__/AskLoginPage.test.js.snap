// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskLoginPage matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center m-4 d-flex align-items-center justify-content-center"
      >
        <div
          style="width: 48rem;"
        >
          <h1>
            Employee Polls
          </h1>
          <img
            alt="login"
            class="image"
            src="loginImage.png"
          />
          <h2>
            Please login to view your polls.
          </h2>
          <button
            class="ripple ripple-surface ripple-surface-light btn btn-primary"
            data-testid="login-button"
            role="button"
          >
            Let's login!
            <div
              class="ripple-wave"
              style="left: 0px; top: 0px; height: 0px; width: 0px; transition-delay: 0s, 250ms; transition-duration: 500ms, 250ms;"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center m-4 d-flex align-items-center justify-content-center"
    >
      <div
        style="width: 48rem;"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          alt="login"
          class="image"
          src="loginImage.png"
        />
        <h2>
          Please login to view your polls.
        </h2>
        <button
          class="ripple ripple-surface ripple-surface-light btn btn-primary"
          data-testid="login-button"
          role="button"
        >
          Let's login!
          <div
            class="ripple-wave"
            style="left: 0px; top: 0px; height: 0px; width: 0px; transition-delay: 0s, 250ms; transition-duration: 500ms, 250ms;"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
